src/eShop.AppHost/Program.cs

```diff
@@ -18,9 +18,14 @@
 
 var launchProfileName = ShouldUseHttpForEndpoints() ? "http" : "https";
 
+// To configure Azure Monitor Application Insights, add the following:
+// var insights = builder.AddConnectionString("myInsightsResource", "APPLICATIONINSIGHTS_CONNECTION_STRING");
+// You also need to add .WithReference(insights) to each service that should be monitored.
+
 // Services
 var identityApi = builder.AddProject<Projects.Identity_API>("identity-api", launchProfileName)
     .WithExternalHttpEndpoints()
+//  .WithReference(insights)    // Sample reference to Azure Monitor Application Insights. Copy/paste for each service.
```

Comment:
I was going to add an overload to have a conditional reference, but it already seems to be there, so the following could be added to each app and *not* need to be commented out. It should silently ignore it if the resource is null.

``` c#
    .WithReference(appInsights,null,true);
```

@mitchdenny - It turns out this doesn't work as the resource parameter isn't nullable. Is there a cleaner workaround?

Reply:
Yeah the .WithReference doesn't like that appInsights could be null.
![image](https://github.com/user-attachments/assets/b70d4218-8297-439e-9bd0-f0dfc5099a44)

Reply:
Why not just use an conditional? I'm not sure we want to invent a new API for this?

Reply:
@mitchdenny conditionals don't fit nicely into the chained builder pattern, unless there is something I don't know?